src/py21cmfast/src/PerturbField.c

```diff
@@ -527,9 +529,9 @@ int ComputePerturbField(
 
         LOG_SUPER_DEBUG("density_perturb: ");
         if(user_params->PERTURB_ON_HIGH_RES){
-            debugSummarizeBoxComplex(HIRES_density_perturb, dimension, user_params->NON_CUBIC_FACTOR, "  ");
+            debugSummarizeBox((float *)HIRES_density_perturb, dimension, dimension, 2*(dimension_z/2 + 1), "  ");
```

Comment:
I'm a little perplexed about the `2*(dimension_z/2+1)` here... is it FFT padding?

Reply:
yeah, comparing `R_INDEX`, `R_FFT_INDEX`, and `C_INDEX` we see that the fft padding adds 2 to the z-dimension on even-dimensioned arrays and 1 to odd, and so the indexing uses `dimension_z`, `(dimension_z / 2 + 1)`  and `2*(dimension_z/2 + 1)` respectively. Since the fftw box is in real-space here, I use the float version with the padded z-index.

https://fftw.org/fftw3_doc/Multi_002dDimensional-DFTs-of-Real-Data.html

Reply:
OK, great!